plugins/x86/x86_tools_prefix.ml

```diff
@@ -3,34 +3,25 @@ open Bap.Std
 open X86_opcode
 open X86_tools_types
 
-module type W = module type of Word
 module Make (RR : RR) (FR : FR) (IV : IV) : PR = struct
-  type t = [
-    | `LOCK_PREFIX
-    | `REP_PREFIX
-    | `REPE_PREFIX
-    | `REPNE_PREFIX
-    | `EXCEPTION of int
-    | `NONE
-  ] [@@deriving sexp, compare]
+  type t = X86_prefix.t [@@deriving sexp, compare]
 
-  let parse mem op =
-    let code = Memory.(mem ^! min_addr mem) |>
+  let parse mem op = 
```

Comment:
I think that the whole code that deals with prefixes is broken. Maybe it follows the spec, but it doesn't follow the real life. Also, since we're trying to make the lifter extensible, this particular place is yet another bottleneck, adding new instruction will eventually require us to change this code.